### ISSUE TITLE
Alpha Surveillance missions can clash with other missions

### ISSUE THREAD

**Issue by ThrawnCA** [endless-sky/endless-sky#7964]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The "Alpha Surveillance G" mission includes the player being shot with a stun weapon and knocked out when they land on Farpoint, but it doesn't prevent other mission triggers from firing. Since the player is supposed to be unconscious and in custody, any other mission events make no sense.

### Steps to Reproduce

1. Start "Alpha Surveillance G" and take Admiral Danforth to Avalon.
2. Return to Hai space and visit spaceports until you trigger the Scar's Legion pirate attack and begin "Hai Reveal: Pirate Troubles [1]".
3. Visit Farpoint to progress both missions.
4. The player first gets knocked out, then visits the Navy base to seek information about Scar's Legion, then wakes up in custody.

### Expected Behavior

When the player is knocked out during "Alpha Surveillance G", all other mission events should be postponed until they wake up and are released. Or, alternatively, the event where the player is knocked out should be postponed until after other events have occurred.

### Operating System

Ubuntu Linux 22.04

### Game Source

Steam

### Game Version

0.9.16.1

**Comment by warp-core** [comment]
Allowing landing missions to be `priority` and making Alpha Surveillance H priority would probably fix this, but I'm not sure what the reason for not letting landing missions be priority in the first place is.

An alternative could be to have some way for the `on complete` action of G to set a condition or something that tells the game not to offer any mission other than H, which would clear that condition in its `on offer` action.

**Comment by ThrawnCA** [comment]
I saw a comment in a mission data file suggesting that mission events are affected by the alphabetical order of their names. Could this mission perhaps have its name tweaked so it runs last? The important thing is that it shouldn't have any other event occur in the _middle_ of it, when the player is meant to be arrested and unconscious.

**Comment by warp-core** [comment]
> I saw a comment in a mission data file suggesting that mission events are affected by the alphabetical order of their names. Could this mission perhaps have its name tweaked so it runs last? The important thing is that it shouldn't have any other event occur in the _middle_ of it, when the player is meant to be arrested and unconscious.

Yes, but I'm not really a fan of trying to control offer order with names.

It would also be that H would have to be named to offer before anything else, since the first conversation is the completion of G, and I believe completion actions will run before any new missions are offered.

**Comment by 1010todd** [comment]
> > I saw a comment in a mission data file suggesting that mission events are affected by the alphabetical order of their names. Could this mission perhaps have its name tweaked so it runs last? The important thing is that it shouldn't have any other event occur in the _middle_ of it, when the player is meant to be arrested and unconscious.
> 
> Yes, but I'm not really a fan of trying to control offer order with names.
> 
> It would also be that H would have to be named to offer before anything else, since the first conversation is the completion of G, and I believe completion actions will run before any new missions are offered.

Instead of just `minor` and `priority` I think we should just have `priority <x>` where `<x>` is number so higher(or lower) priority missions gets offered first or something, this more granularity would allow other missions to use `priority` worrying less about other missions that also uses `priority`.

**Comment by Amazinite** [comment]
> Allowing landing missions to be priority and making Alpha Surveillance H priority would probably fix this, but I'm not sure what the reason for not letting landing missions be priority in the first place is.

We can now make `landing` missions have `priority`. #11416

We could also give the Alpha Surveillance mission a high offer precedence so that it always offers first. #10887
